programs/openbook-v2/src/instructions/consume_events.rs

```diff
@@ -54,43 +55,39 @@ pub fn consume_events(
     let limit = std::cmp::min(limit, MAX_EVENTS_CONSUME);
 
     let mut market = ctx.accounts.market.load_mut()?;
-    let mut event_queue = ctx.accounts.event_queue.load_mut()?;
+    let mut event_heap = ctx.accounts.event_heap.load_mut()?;
     let remaining_accs = &ctx.remaining_accounts;
 
-    let slots_to_consume: Vec<usize> = match slots {
-        Some(slots) => slots
-            .into_iter()
-            .filter(|slot| !event_queue.nodes[*slot].is_free())
-            .take(limit)
-            .collect(),
-        None => event_queue
-            .iter()
-            .map(|(_event, slot)| slot)
-            .take(limit)
-            .collect(),
-    };
+    let slots_to_consume = slots
+        .unwrap_or_default()
+        .into_iter()
+        .filter(|slot| !event_heap.nodes[*slot].is_free())
+        .chain(event_heap.iter().map(|(_event, slot)| slot))
+        .unique()
+        .take(limit)
+        .collect_vec();
```

Comment:
this might be very expensive @skrrb can you check how much this impacts performance?

Reply:
it takes ~8.5k to do chain two vectors, get the unique values and take up to 8 elements which will be the worse case scenario (`MAX_EVENTS_CONSUME = 8`). if the limit is 4, this is reduced to ~3.8